src/xrCore/_math.cpp

```diff
@@ -215,9 +215,49 @@ XRCORE_API u64 QPC() noexcept
     return _dest;
 }
 
+#if defined(XR_ARM)
+#include <sys/syscall.h>
+#include <linux/perf_event.h>
+
+static int fddev = -1;
+__attribute__((constructor)) static void
+init(void)
+{
+    static struct perf_event_attr attr;
+    attr.type = PERF_TYPE_HARDWARE;
+    attr.config = PERF_COUNT_HW_CPU_CYCLES;
+    fddev = syscall(__NR_perf_event_open, &attr, 0, -1, -1, 0);
+}
+
+__attribute__((destructor)) static void
+fini(void)
+{
+    close(fddev);
+}
+#endif
+
 XRCORE_API u64 GetCLK()
 {
-#if defined(XR_ARM64)
+#if defined(XR_ARM)
+    long long result = 0;
+    if (read(fddev, &result, sizeof(result)) < sizeof(result)) return 0;
+    return result;
+/*
+    uint32_t pmccntr;
+    uint32_t pmuseren;
+    uint32_t pmcntenset;
+    // Read the user mode perf monitor counter access permissions.
+    asm volatile("mrc p15, 0, %0, c9, c14, 0" : "=r"(pmuseren));
+    if (pmuseren & 1) {  // Allows reading perfmon counters for user mode code.
+        asm volatile("mrc p15, 0, %0, c9, c12, 1" : "=r"(pmcntenset));
+        if (pmcntenset & 0x80000000ul) {  // Is it counting?
+            asm volatile("mrc p15, 0, %0, c9, c13, 0" : "=r"(pmccntr));
+            // The counter is set up to count every 64th cycle
+            return static_cast<int64_t>(pmccntr) * 64;  // Should optimize to << 6
+        }
+    }
+*/
```

Comment:
What's the purpose of the commented code?

Reply:
I can remove it, if you don't like) It was 1st try and it failed, because of cycle count is restricted from user-mode by default on ARM/Linux. Check here for details:
http://neocontra.blogspot.com/2013/05/user-mode-performance-counters-for.html

Reply:
Yes, it's better to remove it.

Reply:
done